apps/b2b/components/Input.tsx

```diff
@@ -0,0 +1,43 @@
+import classNames from 'classnames'
+import {
+  ClassAttributes,
+  ForwardedRef,
+  forwardRef,
+  HTMLAttributes,
+  HTMLProps,
+} from 'react'
+
+interface InputProps
+  extends HTMLAttributes<HTMLInputElement>,
+    ClassAttributes<HTMLInputElement>,
+    HTMLProps<HTMLInputElement> {
+  type?:
+    | 'text'
+    | 'email'
+    | 'number'
+    | 'password'
+    | 'tel'
+    | 'file'
+    | 'date'
+    | 'search'
+}
+
+const Input = forwardRef(
+  (props: InputProps, ref: ForwardedRef<HTMLInputElement>) => {
+    const { type = 'text', className, ...otherProps } = props
+    return (
+      <input
+        ref={ref}
+        type={type}
+        className={classNames(
+          'block w-full border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 focus:outline-none focus:ring-0 h-10',
+          className
+        )}
+        {...otherProps}
+      />
+    )
+  }
+)
+
+Input.displayName = 'Input'
```

Comment:
since we're heavily tied up to react hook forms, and since they're now pushing to just pass a destructured register, how about just passing register as props to the component instead of creating a forwardRef component?

Reply:
will check this out but wouldn't we still need an Input with forwardedRef? maybe we could create an Input specialized to Forms and another Input for normal cases?

Reply:
basically, just pass `register={register}` to Input. make register a prop of Input

Reply:
on the actual input, can just use `<input {...register(name)}>` instead